server/routes/reports/createReport.test.ts

```diff
@@ -0,0 +1,270 @@
+import type { Express } from 'express'
+import request, { type Agent, type Response } from 'supertest'
+
+import format from '../../utils/format'
+import { appWithAllRoutes } from '../testutils/appSetup'
+import { type ErrorResponse, IncidentReportingApi } from '../../data/incidentReportingApi'
+import { convertReportWithDetailsDates } from '../../data/incidentReportingApiUtils'
+import { mockReport } from '../../data/testData/incidentReporting'
+import { mockThrownError } from '../../data/testData/thrownErrors'
+
+jest.mock('../../data/incidentReportingApi')
+
+let app: Express
+let incidentReportingApi: jest.Mocked<IncidentReportingApi>
+
+beforeEach(() => {
+  app = appWithAllRoutes({})
+  incidentReportingApi = IncidentReportingApi.prototype as jest.Mocked<IncidentReportingApi>
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+function expectOnStep1(res: Response): void {
+  expect(res.request.url.endsWith('/create-report')).toBeTruthy()
+  expect(res.text).toContain('Select incident type')
+}
+
+function expectOnStep2(res: Response): void {
+  expect(res.request.url.endsWith('/create-report/details')).toBeTruthy()
+  expect(res.text).toContain('Incident details')
+}
+
+describe('Creating a report', () => {
+  let agent: Agent
+
+  beforeEach(() => {
+    agent = request.agent(app)
+  })
+
+  describe('Step 1: selecting incident type', () => {
+    it('should list incident types that can be selected', () => {
+      return agent
+        .get('/create-report')
+        .expect(200)
+        .expect(res => {
+          expectOnStep1(res)
+          expect(res.text).not.toContain('There is a problem')
+
+          // active type
+          expect(res.text).toContain('BOMB_THREAT')
+          expect(res.text).toContain('Bomb threat')
+          // inactive type
+          expect(res.text).not.toContain('OLD_DRUGS')
+          expect(res.text).not.toContain('Drugs')
+        })
+    })
+
+    it('should show an error message if submitted without choosing a type', () => {
+      return agent
+        .post('/create-report')
+        .send({})
+        .redirects(1)
+        .expect(200)
+        .expect(res => {
+          expectOnStep1(res)
+          expect(res.text).toContain('There is a problem')
+          expect(res.text).toContain('Choose one of the options')
+        })
+    })
+
+    it('should proceed to step 2 if an incident type was chosen', () => {
+      return agent
+        .post('/create-report')
+        .send({ type: 'DISORDER' })
+        .redirects(1)
+        .expect(200)
+        .expect(res => {
+          expectOnStep2(res)
+        })
+    })
+  })
+
+  describe('Step 2: entering date and description', () => {
+    beforeEach(() => {
+      return agent.post('/create-report').send({ type: 'DISORDER' })
+    })
+
+    const incidentDateAndTime = new Date('2024-10-21T16:32:00+01:00')
+    const validPayload = {
+      incidentDate: '21/10/2024',
+      '_incidentTime-hours': '16',
```

Comment:
I'm not so sure about this naming convention, these are not private fields or anything, but I guess maybe there is some value in distinguishing them from "normal" fields? Unless we want to render them separately from the usual `for (field of fields)` loop

Reply:
we can't render them separately: it's the owning `incidentTime` field that is used in the template